File: models/collection.go
package models

type Collection struct {
	ID        int64  `json:"id"`
	ProjectID int64  `json:"project_id"`
	ParentID  *int64 `json:"parent_id"`
	Name      string `json:"name"`
	SortOrder int    `json:"sort_order"`
	CreatedAt string `json:"created_at"`
	UpdatedAt string `json:"updated_at"`
}

type TreeItem struct {
	ID        int64      `json:"id"`
	Name      string     `json:"name"`
	Type      string     `json:"type"`
	Method    string     `json:"method,omitempty"`
	URL       string     `json:"url,omitempty"`
	Children  []TreeItem `json:"children"`
	SortOrder int        `json:"sort_order"`
}

File: frontend/src/stores/collection.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'
import type { TreeItem } from '../types/collection'
import { GetCollectionTree, CreateCollection, CreateRequest } from '../../wailsjs/go/main/App'

export const useCollectionStore = defineStore('collection', () => {
  const tree = ref<TreeItem[]>([])
  const expandedKeys = ref<number[]>([])

  async function loadTree(projectId: number | null) {
    if (!projectId) {
      tree.value = []
      return
    }
    try {
      const raw = await GetCollectionTree(projectId)
      tree.value = (raw || []) as any as TreeItem[]
    } catch {
      tree.value = []
    }
  }

  async function createCollection(projectId: number, parentId: number | null, name: string) {
    await CreateCollection(projectId, parentId as any, name)
    await loadTree(projectId)
  }

  async function createRequestInCollection(projectId: number, collectionId: number, name: string, method: string) {
    await CreateRequest(collectionId, name, method)
    await loadTree(projectId)
  }

  return { tree, expandedKeys, loadTree, createCollection, createRequestInCollection }
})
